game/src/AI/States/SearchRadius.gd

```diff
@@ -0,0 +1,23 @@
+extends State
+"""
+State machine state that waits for the player to arrive to initiate a charging attack.
+Sets the target position to aim for to the straight line charge behavior.
+"""
+
+func enter(msg: Dictionary = {}) -> void:
+	$"../../SearchRadius".connect("body_entered", self, "_on_Area2D_body_entered")
```

Comment:
Don't get ancestors like that. You can use the `owner` to store that node as a dependency and access it from the state as e.g. `owner.search_area`